src/shims/unix/fs.rs

```diff
@@ -600,8 +594,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         } else if whence == this.eval_libc_i32("SEEK_END") {
             SeekFrom::End(i64::try_from(offset).unwrap())
         } else {
-            this.set_last_error(LibcError("EINVAL"))?;
-            return Ok(Scalar::from_i64(-1));
```

Comment:
This returns an `i64`, not an `i32`, so your patch introduces an ICE here! So please add a new `set_last_error_and_return_i64` instead and use that here, or leave this to a future PR.

Please make sure to always maintain the original type, it's quite important to not screw this up and because this is "dynamic" (it's types of the interpreted program, not of the interpreter), even `./miri check` will not catch this.